Blog post: Announcing MyST-JS

choldgraf opened the pull request:
This is a blog post announcing MyST as a standalone project from Jupyter Book, as well as announcing the recent MyST-JS efforts that @rowanc1 @ co have been leading. Its goal is to highlight major aspects of the work we've been doing over the past few months, and give an idea of how these projects will continue to evolve moving forward.

It touches on some of the discussions in https://github.com/executablebooks/meta/issues/838 as well as #493 , though not sure it closes either of them. I'll leave it to @rowanc1 to decide when #838 can be closed.

Please provide feedback and suggestions on the content! (cc in particular @rowanc1 since Curvenote is mentioned in this post, as well as @gregcaporaso and @jstac who have been involved in setting up the collaboration over the past several months).

### To do

- [x] Make js.myst-tools.org work - some of the shortcuts are still here (js.myst.tools needs to be js.myst-tools.org).
- [x] Any more calls to action or social media-style links?
    - Social handle ([@myst_tools](https://twitter.com/myst_tools))
    - Mailing list (updates.myst-tools.org)


### Other things that are changed

I took the opportunity to make a few other updates to our blog:

- We now use `dirhtml`
- The root is now `blog` instead of `updates`
- I made a few updates to our `tools` page to reflect new tools and links.

This will break old links to our blog, but it was non-trivial to redirect if we use `dirhtml` now. However I figured that because we haven't had many posts historically, it wouldn't be a huge problem if we had a one-time break. Let me know if anybody strongly opposes that though.

fperez commented:
Thx @choldgraf!! I have no further edits at this poin.

rowanc1 commented on docs/blog/2023/announce-mystjs.md:
```diff
@@ -0,0 +1,143 @@
+
+By bringing MyST into JavaScript, we can integrate it directly with JupyterLab and other web-based Jupyter interfaces.
+We have already explored a few ways of doing this via JupyterLab extensions.
+The [`jupyterlab-myst` repository](https://github.com/executablebooks/jupyterlab-myst) brings basic functionality of MyST in JupyterLab via `markdown-it` plugins, and the [`jupyterlab-mystjs`](https://github.com/executablebooks/jupyterlab-mystjs) repository is a more fully-functional document engine with MyST in JupyterLab.
+We hope to merge the second repository into the first in the future, so follow [`jupyterlab-myst`](https://github.com/executablebooks/jupyterlab-myst) moving forward.
```
Depending on when this gets released - @agoose77 and I are having a chat tomorrow to potentially (hopefully!) bring the two efforts together!

rowanc1 replied:
This is being merged! See https://github.com/executablebooks/jupyterlab-myst/issues/57 and next comment!

rowanc1 commented on docs/blog/2023/announce-mystjs.md:
```diff
@@ -0,0 +1,143 @@
+It also has major functionality that is missing in MyST-JS (for example, full internationalization support and extensions).
+We will continue to improve and maintain this stack (both as individual projects in the Sphinx ecosystem and as the Jupyter Book distribution).
+
+The MyST-JS project will initially focus its efforts on a slightly different use-case: **scientific publishing usecases**, such as reproducible analyses meant for publication, and integration with publisher workflows like [JATS](https://en.wikipedia.org/wiki/Journal_Article_Tag_Suite) and [Latex](https://www.latex-project.org/).
```
```suggestion
The MyST-JS project will initially focus its efforts on a slightly different use-case: **scientific publishing usecases**, such as reproducible analyses meant for publication, and integration with publisher workflows like [JATS](https://en.wikipedia.org/wiki/Journal_Article_Tag_Suite) and [LaTeX](https://www.latex-project.org/).
```

:)

rowanc1 commented on docs/tools.md:
```diff
@@ -1,13 +1,39 @@
+MyST Markdown is a language- and implementation-agnostic flavor of markdown with support from several tools (some of which are stewarded by us and listed below).
+
+- [`myst-tools.org`](https://myst-tools.org) is a landing page about the project.
+- [`spec.myst.tools.org`](https://spec.myst-tools.org) describes the MyST specification.
```
```suggestion
- [`spec.myst-tools.org`](https://spec.myst-tools.org) describes the MyST specification.
```

rowanc1 commented on docs/blog/2023/announce-mystjs.md:
```diff
@@ -0,0 +1,143 @@
+
+The [MyST-JS project](https://github.com/executablebooks/mystjs) is written entirelly in TypeScript, with the goal of being re-usable across more applications and interfaces.
+It also allows us to leverage the vast ecosystem of JavaScript libraries and frameworks for communication.
+For example, we can use [templating frameworks to export MyST documents to a variety of LaTeX publication formats](https://github.com/myst-templates).
+In the future, we hope to leverage the [unifiedjs project](https://unifiedjs.com/) to make MyST documents even more structured and consumable as data.
```
```suggestion
For example, we can use [templating frameworks to export MyST documents to a variety of LaTeX publication formats](https://github.com/myst-templates), and leverage the [unifiedjs project](https://unifiedjs.com/) to make MyST documents even more structured and consumable as data.
```

We are already making considerable use of unifiedjs (e.g. all transformations of the AST, parsing latex, serializing tex), we aren't making use of it yet for markdown-parsing which I think is what the future part referred to. I just tweaked the wording slightly as it is all future facing already.

rowanc1 commented on docs/blog/2023/announce-mystjs.md:
```diff
@@ -0,0 +1,143 @@
+Over the years we have seen interest in MyST raise considerably - it allows authors and researchers to express themselves more efficiently than traditional source text languages like LaTeX.
+MyST has also become [a staple in the Python documentation ecosystem](https://www.sphinx-doc.org/en/master/usage/markdown.html), and is the backbone of authoring content in [Jupyter Book](https://jupyterbook.org).
+
+## Bringing MyST to JavaScript with a new partnership
```
🚀

rowanc1 reviewed (dismissed):
This is awesome, thanks @choldgraf for putting this together and taking a pass on the whole website. Super excited to elevate MyST over the coming years - and so excited to have Curvenote be a partner of the ExecutableBooks community. 🚀

jstac commented:
Great job @choldgraf .  It reads very well.  I gave it another pass through and found no typos.

gregcaporaso commented on docs/blog/2023/announce-mystjs.md:
```diff
@@ -0,0 +1,143 @@
+
+### MyST will have first-class support in JavaScript
+
+The [MyST-JS project](https://github.com/executablebooks/mystjs) is written entirelly in TypeScript, with the goal of being re-usable across more applications and interfaces.
```
```suggestion
The [MyST-JS project](https://github.com/executablebooks/mystjs) is written entirely in TypeScript, with the goal of being re-usable across more applications and interfaces.
```

gregcaporaso reviewed (dismissed):
This looks great @choldgraf!

rowanc1 commented on docs/blog/2023/announce-mystjs.md:
```diff
@@ -0,0 +1,143 @@
+
+### MyST will have a Jupyter Lab extension
+
+By bringing MyST into JavaScript, we can integrate it directly with JupyterLab and other web-based Jupyter interfaces.
+We have already explored a few ways of doing this via JupyterLab extensions.
+The [`jupyterlab-myst` repository](https://github.com/executablebooks/jupyterlab-myst) brings basic functionality of MyST in JupyterLab via `markdown-it` plugins, and the [`jupyterlab-mystjs`](https://github.com/executablebooks/jupyterlab-mystjs) repository is a more fully-functional document engine with MyST in JupyterLab.
+We hope to merge the second repository into the first in the future, so follow [`jupyterlab-myst`](https://github.com/executablebooks/jupyterlab-myst) moving forward.
```
@agoose77 and I have worked today to get these two repositories merged together, we are tracking all outstanding pieces in https://github.com/executablebooks/jupyterlab-myst/issues/57 and will have this fully deployed in the next day or so!

```suggestion
By bringing MyST into JavaScript, we can integrate it directly with JupyterLab and other web-based Jupyter interfaces.
For example, the [`jupyterlab-myst`](https://github.com/executablebooks/jupyterlab-myst) extension brings native MyST rendering into JupyterLab including admonitions, cross-references, and figure numbering using `mystjs` as the document engine.
```

agoose77 commented on docs/blog/2023/announce-mystjs.md:
```diff
@@ -0,0 +1,143 @@
+You can find the documentation for MyST JS at [js.myst-tools.org](https://myst-tools.org/docs/mystjs). You can also find an [issue from Rowan discussing major new planned functionality here](https://github.com/executablebooks/meta/issues/838).
+:::
+
+### MyST will have a Jupyter Lab extension
```
```suggestion
### MyST will have a JupyterLab extension
```

choldgraf commented:
OK I fixed a few links and also added links to the two marquee projects on our landing page. Will merge this one now and send out a tweet about it - @rowanc1 cc in case you wanna tweet via `@myst-tools` as well.

choldgraf commented:
Post is live! Here's a tweet about it:

https://twitter.com/choldgraf/status/1625170491639054340

link to the post:

https://executablebooks.org/en/latest/blog/2023/announce-mystjs/

please share any messages that you'd like others to signal-boost!

fperez commented:
[Done](https://twitter.com/fperez_org/status/1625174969255874561)! Great job @choldgraf & team, so happy to see this grow!
